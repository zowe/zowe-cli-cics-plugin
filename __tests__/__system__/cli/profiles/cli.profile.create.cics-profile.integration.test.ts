/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript, isStderrEmptyForProfilesCommand } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/doc/ITestPropertiesSchema";

let testEnvironment: ITestEnvironment<ITestPropertiesSchema>;

/*
 NOTE!!  The cics plugin must be installed into Brightside in order to test the creation of a cics profile.
  */

describe("Create cics Profile", () => {

    const args: any[] = [];

    // Create the unique test environment
    beforeAll(async () => {
        testEnvironment = await TestEnvironment.setUp({
            testName: "zos_create_cics_profile",
            installPlugin: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(testEnvironment);
    });

    describe("Success scenarios", () => {

        it("should display create cics profile help", () => {

            const scriptPath = __dirname + "/__scripts__/create_cics_profile_help.sh";
            const response = runCliScript(scriptPath, testEnvironment, args);

            expect(response.stderr.toString()).toBe("");
            expect(response.status).toBe(0);
            expect(response.stdout.toString()).toMatchSnapshot();
        });

        it("should create cics profile", () => {

            const scriptPath = __dirname + "/__scripts__/create_cics_profile.sh";
            const response = runCliScript(scriptPath,
                testEnvironment, args);

            expect(isStderrEmptyForProfilesCommand(response.stderr)).toBe(true);
            expect(response.status).toBe(0);
            expect(response.stdout.toString()).toContain("Profile created successfully");
        });
    });
});
