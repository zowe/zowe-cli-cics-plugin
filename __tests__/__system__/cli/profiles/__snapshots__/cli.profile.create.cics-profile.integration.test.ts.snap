// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create cics Profile Success scenarios should display create cics profile help 1`] = `
"===============CREATE cics PROFILE HELP===============

 COMMAND NAME
 ------------

   cics-profile | cics

 DESCRIPTION
 -----------

   A cics profile is required to issue commands in the cics command group that
   interact with CICS regions. The cics profile contains your host, port, user
   name, and password for the IBM CICS management client interface (CMCI) server of
   your choice.

 USAGE
 -----

   zowe profiles create cics-profile <profileName> [options]

 POSITIONAL ARGUMENTS
 --------------------

   profileName		 (string)

      Specifies the name of the new cics profile. You can load this profile by using
      the name on commands that support the \\"--cics-profile\\" option.

 REQUIRED OPTIONS
 ----------------

   --host  | -H (string)

      The CMCI server host name

   --user  | -u (string)

      Your username to connect to CICS

   --password  | -p (string)

      Your password to connect to CICS

 OPTIONS
 -------

   --port  | -P (number)

      The CMCI server port

      Default value: 1490

   --region-name  (string)

      The name of the CICS region name to interact with

   --cics-plex  (string)

      The name of the CICSPlex to interact with

   --overwrite  | --ow (boolean)

      Overwrite the cics profile when a profile of the same name exists.

 CICS CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --protocol  | -o (string)

      Specifies CMCI protocol (http or https).

      Default value: http
      Allowed values: http, https

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - Create a cics profile named 'cics123' to connect to CICS at
   host zos123 and port 1490:

      $ zowe profiles create cics-profile cics123 --host zos123 --port 1490 --user ibmuser --password myp4ss

"
`;
